app/sagas/init.js

```diff
@@ -29,71 +22,15 @@ export const initLocalSettings = function* initLocalSettings() {
 
 const restore = function* restore() {
 	try {
-		let hasMigration;
-		if (isIOS) {
-			hasMigration = yield AsyncStorage.getItem('hasMigration');
-		}
-
-		let { token, server } = yield all({
-			token: RNUserDefaults.get(RocketChat.TOKEN_KEY),
-			server: RNUserDefaults.get('currentServer')
+		const { token, server } = yield all({
+			token: UserPreferences.getStringAsync(RocketChat.TOKEN_KEY),
```

Comment:
I'm not sure if it's related to the error below, but sometimes the app never resolves this function and the app hangs on splash screen.

Reply:
Yeah, after my changes, it doesn't happen again for me.